lib/src/layer/tile_layer/tile_layer.dart

```diff
@@ -0,0 +1,549 @@
+        if (options.overrideTilesWhenUrlChanges) {
+          _tileManager.reloadImages(options, _wrapX, _wrapY);
+        } else {
+          reloadTiles = true;
+        }
+      }
+    }
+
+    if (reloadTiles) {
+      _tileManager.removeAll(options.evictErrorTileStrategy);
+      _resetView();
+      _update(null);
+    }
+  }
+
+  void _initThrottleUpdate() {
+    if (options.updateInterval == null) {
+      _throttleUpdate = null;
+    } else {
+      _throttleUpdate = StreamController<LatLng?>(sync: true);
+      _throttleUpdate!.stream
+          .transform(
+            util.throttleStreamTransformerWithTrailingCall<LatLng?>(
+              options.updateInterval!,
+            ),
+          )
+          .listen(_update);
+    }
+  }
+
+  @override
+  void dispose() {
+    _tileManager.removeAll(options.evictErrorTileStrategy);
+    _resetSub?.cancel();
+    _moveSub?.cancel();
+    _pruneLater?.cancel();
+    options.tileProvider.dispose();
+    _throttleUpdate?.close();
+
+    super.dispose();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder(
+      stream: widget.stream,
+      builder: (context, snapshot) {
+        final tilesToRender = _tileZoom == null
+            ? _tileManager.all()
+            : _tileManager.sortedByDistanceToZoomAscending(
+                options.maxZoom, _tileZoom!);
+
+        final Map<double, TileTransformation> zoomToTransformation = {};
+
+        final tileWidgets = <Widget>[
+          for (var tile in tilesToRender)
+            TileWidget(
+              tile: tile,
+              size: _tileSize,
+              tileTransformation: zoomToTransformation[tile.coords.z] ??
+                  (zoomToTransformation[tile.coords.z] =
+                      _transformationCalculator.transformationFor(
+                    tile.coords.z,
+                    map,
+                  )),
+              errorImage: options.errorImage,
+              tileBuilder: options.tileBuilder,
+              key: ValueKey(tile.coordsKey),
+            )
+        ];
+
+        final tilesContainer = Stack(
+          children: tileWidgets,
+        );
+
+        final tilesLayer = options.tilesContainerBuilder == null
+            ? tilesContainer
+            : options.tilesContainerBuilder!(
+                context,
+                tilesContainer,
+                tilesToRender,
+              );
+
+        final attributionLayer =
+            widget.options.attributionBuilder?.call(context);
+
+        return Opacity(
+          opacity: options.opacity,
+          child: Container(
+            color: options.backgroundColor,
+            child: Stack(
+              alignment: widget.options.attributionAlignment,
+              children: [
+                tilesLayer,
+                if (attributionLayer != null) attributionLayer,
+              ],
+            ),
+          ),
+        );
+      },
+    );
+  }
+
+  CustomPoint getTileSize() {
+    return _tileSize;
+  }
```

Comment:
Removing unnecessary statements.

```suggestion
  CustomPoint getTileSize() => _tileSize;
```